ostd/src/arch/x86/cpu/mod.rs

```diff
@@ -76,6 +77,27 @@ impl UserContext {
     pub fn fp_regs_mut(&mut self) -> &mut FpRegs {
         &mut self.fp_regs
     }
+
+    /// Sets thread-local storage pointer.
+    pub fn set_tls_pointer(&mut self, tls: usize) {
+        self.set_fsbase(tls)
+    }
+
+    /// Gets thread-local storage pointer.
+    pub fn tls_pointer(&self) -> usize {
+        self.fsbase()
+    }
+
+    /// Activates thread-local storage pointer.
```

Comment:
```
/// Activates thread-local storage pointer on the current CPU.
///
/// # Safety
///
/// The method by itself is safe because the value of the TLS register won't affect kernel code.
/// But if the user relies on the TLS pointer, make sure that the pointer is correctly set when entering
/// the user space. 
```